### PR TITLE
[Patitioner] Refactor the initializing partition code.

### PR DESCRIPTION
**Pull request by beicy**
*Description*:
When we create the initial partitions, we can combine the level cut and mapping together.  In addition, add the function to remove duplicate memory calculating.

*Testing*:
ninja check. restnet-runtime.
*Documentation*:
[Optional Fixes #issue]

Please see a detailed explanation of how to fill out the fields in the relevant sections in PULL_REQUEST.md.

### CODE REVIEW

File: lib/Partitioner/Partitioner.cpp [hunk 1]
```diff
@@ -19,6 +19,22 @@
 using namespace glow;
 using llvm::isa;
 
+/// Check if the memory of \p node inputs is calculated already. \p return the
+/// total used memory after \p node is considered.
+static uint64_t updateUsedMem(std::set<Storage *> &usedStorage,
```
Comment (rdzhabarov) [inline lib/Partitioner/Partitioner.cpp]: from the interface perspective usedStorage should be const.

File: lib/Partitioner/Partitioner.cpp [hunk 2]
```diff
@@ -19,6 +19,22 @@
 using namespace glow;
 using llvm::isa;
 
+/// Check if the memory of \p node inputs is calculated already. \p return the
```
Comment (nadavrot) [inline lib/Partitioner/Partitioner.cpp]: \p stands for "parameter". No need for \p before the return. Use "\returns". 

Reference and examples here:

http://www.doxygen.nl/manual/docblocks.html

File: lib/Partitioner/Partitioner.cpp [hunk 3]
```diff
@@ -19,6 +19,22 @@
...
+/// total used memory after \p node is considered.
+static uint64_t updateUsedMem(const std::set<Storage *> usedStorage,
```
Comment (nadavrot) [inline lib/Partitioner/Partitioner.cpp]: Please pass the set as reference.

**Comment by rdzhabarov** [comment]
> ci/circleci: DEBUG

This is unfortunately got flaky with the latest upgrade to llvm8. @mortzur is looking into this.
Let's merge this once travis is done, i'll put this into fbcode after that.

**Comment by beicy** [comment]
> > ci/circleci: DEBUG
> 
> This is unfortunately got flaky with the latest upgrade to llvm8. @mortzur is looking into this.
> Let's merge this once travis is done, i'll put this into fbcode after that.

Thanks for letting me know! I was trying to figure out why it failed the ci...

File: lib/Partitioner/Partitioner.cpp (hunk 1 above)
Comment (rdzhabarov) [reply lib/Partitioner/Partitioner.cpp]: const std::set<Storage *> &usedStorage

**Pull request merged**